doc/modules/language-guide/pages/basic-concepts.adoc

```diff
@@ -462,14 +462,82 @@ P.unreachable()
 As in the situations above, this function type-checks in all contexts, and when evaluated, traps in all contexts.
 
 [[overview-traps]]
-=== Execution traps stop the program
+=== Traps due to execution failure
 
-Each form above is a simple wrapper around the always-fail use of the link:language-manual{outfilesuffix}#exp-assert[assert primitive]:
+Some errors, such as division by zero, out-of-bounds array indexing, and pattern match failure are not prevented by the type system, but can cause dynamic failures called _traps_.
 
 [source, motoko]
 ....
-assert false
+1/0; // traps due to division by 0
 ....
 
-Dynamically, we call this program-halting behavior a _program(-generated) trap_, and we say that the program _traps_ when it executes this code.
-It will cease to progress further.
+[source, motoko]
```

Comment:
I was surprised to see the text below in this PR.  Is it being truly added, or just moved from some other file?

(Why is it important to have it in this PR, which is mostly about copy editing for terminology?)

Reply:
It's being added because it was no longer accurate and didn't mention trap. Might as well do somehting useful while rearranging the deck chairs.